pkg/codegen/nodejs/gen.go

```diff
@@ -2788,6 +2815,51 @@ func (mod *modContext) genUtilitiesFile(w io.Writer) error {
 	}
 
 	_, err = fmt.Fprintf(w, "%s", code)
+	if err != nil {
+		return err
+	}
+
+	if def.Parameterization != nil {
+
+		parameterValue := fmt.Sprintf("Uint8Array.from(atob(%q), c => c.charCodeAt(0))", base64.StdEncoding.EncodeToString(def.Parameterization.Parameter))
+
+		_, err = fmt.Fprintf(w, `
+const _packageLock = new mutex.Mutex();
```

Comment:
Why do we need the lock?

Reply:
Strictly we don't, the engine will be ok with multiple RegisterPackage calls for the same data in parallel and we lock in the engine. But this saves us doing multiple RegisterPackage calls if a load of resources get registered at the same time.
Is it worth it.... maybe not? Probably depends on how big the parameter value being sent over is.